### ISSUE TITLE
Question about item, cate, pos features combination

### ISSUE THREAD

**Issue by rejae** [zhouxy1003/TIN#1]
`        position = tf.get_variable("pos", shape=[431, hidden_units])  # time embedding 431 buckets
        item = tf.get_variable("item", [item_count, hidden_units // 2])
        cate = tf.get_variable("cate", [cate_count, hidden_units // 2])  # item_emb_w
        item_b = tf.get_variable("item_b", [item_count],
                                 initializer=tf.constant_initializer(0.0))
        cate_list = tf.convert_to_tensor(cate_list, dtype=tf.int64)

        ic = tf.gather(cate_list, self.i)
        i_b = tf.gather(item_b, self.i)
        i_emb = tf.concat(values=[
            tf.nn.embedding_lookup(item, self.i),
            tf.nn.embedding_lookup(cate, ic),
        ], axis=1)
        i_pos_emb = tf.nn.embedding_lookup(position, self.pos)
        i_emb = tf.add(i_emb, i_pos_emb)  # [bs, d]`

Is tf.concat(item,cate) then tf.add(pos) better than tf.concat(item,cate,pos) ?

**Comment by junwei-pan** [comment]
Feel free to try. Your idea shares the spirit of the paper "Rethinking Positional Encoding in Language Pre-Training," that is, to disentangle the item, cate, and pos embeddings and avoid interactions between these three features in the attention as well as representation. This may lead to better performance. Let us know if so.
